plugins/chat/spec/components/chat/mailer_spec.rb

```diff
@@ -1,334 +1,282 @@
-  fab!(:chatters_group) { Fabricate(:group) }
-  fab!(:sender) { Fabricate(:user, group_ids: [chatters_group.id], refresh_auto_groups: true) }
-  fab!(:user_1) do
-    Fabricate(
-      :user,
-      group_ids: [chatters_group.id],
-      last_seen_at: 15.minutes.ago,
-      refresh_auto_groups: true,
-    )
-  end
-  fab!(:chat_channel) { Fabricate(:category_channel) }
-  fab!(:chat_message) { Fabricate(:chat_message, user: sender, chat_channel: chat_channel) }
-  fab!(:user_1_chat_channel_membership) do
-    Fabricate(
-      :user_chat_channel_membership,
-      user: user_1,
-      chat_channel: chat_channel,
-      last_read_message_id: nil,
-    )
-  end
-  fab!(:private_chat_channel) do
-    result =
-      Chat::CreateDirectMessageChannel.call(
-        guardian: sender.guardian,
-        target_usernames: [sender.username, user_1.username],
-      )
-    service_failed!(result) if result.failure?
-    result.channel
+  fab!(:user) { Fabricate(:user, last_seen_at: 1.hour.ago) }
+  fab!(:other) { Fabricate(:user) }
+
+  fab!(:group) do
+    Fabricate(:group, mentionable_level: Group::ALIAS_LEVELS[:everyone], users: [user, other])
   end
 
-  before do
-    SiteSetting.chat_enabled = true
-    SiteSetting.chat_allowed_groups = chatters_group.id
+  fab!(:followed_channel) { Fabricate(:category_channel) }
+  fab!(:non_followed_channel) { Fabricate(:category_channel) }
+  fab!(:muted_channel) { Fabricate(:category_channel) }
+  fab!(:unseen_channel) { Fabricate(:category_channel) }
+  fab!(:direct_message) { Fabricate(:direct_message_channel, users: [user, other]) }
 
-    Fabricate(:user_chat_channel_membership, user: sender, chat_channel: chat_channel)
-  end
+  fab!(:job) { :user_email }
+  fab!(:args) { { type: :chat_summary, user_id: user.id, force_respect_seen_recently: true } }
 
-  def assert_summary_skipped
-    expect(
-      job_enqueued?(job: :user_email, args: { type: "chat_summary", user_id: user_1.id }),
-    ).to eq(false)
+  before do
+    SiteSetting.chat_enabled = true
+    SiteSetting.chat_allowed_groups = Group::AUTO_GROUPS[:everyone]
   end
 
-  def assert_only_queued_once
-    expect_job_enqueued(job: :user_email, args: { type: "chat_summary", user_id: user_1.id })
+  def expect_enqueued
+    expect_enqueued_with(job:, args:) { described_class.send_unread_mentions_summary }
     expect(Jobs::UserEmail.jobs.size).to eq(1)
   end
 
-  describe "for chat mentions" do
-    fab!(:notification) do
-      Fabricate(:notification, notification_type: Notification.types[:chat_mention], user: user_1)
-    end
-    fab!(:mention) do
-      Fabricate(:user_chat_mention, chat_message: chat_message, notifications: [notification])
-    end
+  def expect_not_enqueued
+    expect_not_enqueued_with(job:, args:) { described_class.send_unread_mentions_summary }
+  end
 
-    it "skips users without chat access" do
-      chatters_group.remove(user_1)
+  # This helper is much faster than `Fabricate(:chat_message_with_service, ...)`
```

Comment:
This might be controversial, but using services was awkward (they require jobs to run immediately, but in these tests, we need jobs to be stopped) and they are 10x as slow as "manually" fabricating the notifications.